src/JSONLDResolver.js

```diff
@@ -67,9 +68,36 @@ export default class JSONLDResolver {
     // Expand the property to a full IRI
     const context = await this._context;
     const expandedProperty = context.expandTerm(property, true);
-    if (!ContextUtil.isValidIri(expandedProperty))
+    if (ContextUtil.isValidIri(expandedProperty))
+      return namedNode(expandedProperty);
+
+    let algebra;
+    const prefixes = {};
+    for (const key in context.contextRaw) {
+      if (typeof context.contextRaw[key] === 'string')
+        prefixes[key] = context.contextRaw[key];
+    }
+    // Wrap inside try/catch as 'translate' throws error on invalid paths
+    try {
+      algebra = translate(`SELECT * WHERE { ?s ${expandedProperty} ?o }`, {
+        prefixes,
+      });
+    }
+    catch (e) {
       throw new Error(`The JSON-LD context cannot expand the '${property}' property`);
-    return namedNode(expandedProperty);
+    }
+
+    if (algebra.input.type === 'path') {
+      const query = toSparql(algebra);
+      return query.slice(25, query.length - 7);
```

Comment:
Let's not have magic numbers

Reply:
Would you prefer `.replace(/^SELECT \* WHERE \{ \?s \/, '')`, `.slice('SELECT * WHERE { ?s '.length,)` or something else? Would be cleaner if `toSparql` could be run on `algebra.input`; do you know if there are any internals in that library to do so?

Reply:
The replace would be nice; then we know if things fail.